tools/src/main/java/it/eng/tools/model/ApplicationProperty.java

```diff
@@ -0,0 +1,209 @@
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+
+import it.eng.tools.exception.ApplicationPropertyErrorException;
+import jakarta.validation.ConstraintViolation;
+import jakarta.validation.Validation;
+import jakarta.validation.ValidationException;
+
+@Getter
+@JsonDeserialize(builder = ApplicationProperty.Builder.class)
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+@JsonPropertyOrder(value = {IConstants.ID, IConstants.KEY, IConstants.VALUE, IConstants.SAMPLE_VALUE, IConstants.MANDATORY}, alphabetic = true)
+@Document(collection = "application_properties")
+public class ApplicationProperty {
+
+	/*
+	 * @JsonProperty(IConstants.ID)
+	 * 
+	 * @Setter
+	 * 
+	 * @Id private String id;
+	 */
+
+	@Id
+	@JsonProperty(IConstants.KEY)
+	private String key;
+
+	@JsonProperty(IConstants.VALUE)
+	private String value;
+
+	@JsonProperty(IConstants.SAMPLE_VALUE)
+	private String sampleValue;
+
+	@JsonProperty(IConstants.MANDATORY)
+	private boolean mandatory;
+
+	@CreatedDate
+	private Instant issued;
+
+	@LastModifiedDate
+	private Instant modified;
+
+	@JsonIgnore
+	@CreatedBy
+	private String createdBy;
+
+	@JsonIgnore
+	@LastModifiedBy
+	private String lastModifiedBy;
+
+	@JsonIgnore
+	@Version
+	@Field("version")
+	private Long version;	
+
+	@JsonPOJOBuilder(withPrefix = "")
+	@JsonIgnoreProperties(ignoreUnknown = true)
+	public static class Builder {
+		private ApplicationProperty property;
+
+		private Builder() {
+			property = new  ApplicationProperty();
+		}
+
+		public static Builder newInstance() {
+			return new Builder();
+		}
+
+//		@JsonProperty(IConstants.ID)
+//		public Builder id(String id) {
+//			property.id = id;
+//			return this;
+//		}
+
+		@JsonProperty(IConstants.KEY)
+		public Builder key(String key) {
+			property.key = key;
+			return this;
+		}
+
+		@JsonProperty(IConstants.VALUE)
+		public Builder value(String value) {
+			property.value = value;
+			return this;
+		}
+
+		@JsonProperty(IConstants.SAMPLE_VALUE)
+		public Builder sampleValue(String sampleValue) {
+			property.sampleValue = sampleValue;
+			return this;
+		}
+
+		@JsonProperty(IConstants.MANDATORY)
+		public Builder mandatory(boolean mandatory) {
+			property.mandatory = mandatory;
+			return this;
+		}
+
+		@JsonProperty("createdBy")
+		public Builder createdBy(String createdBy) {
+			property.createdBy = createdBy;
+			return this;
+		}
+
+		@JsonProperty("lastModifiedBy")
+		public Builder lastModifiedBy(String lastModifiedBy) {
+			property.lastModifiedBy = lastModifiedBy;
+			return this;
+		}
+
+		public Builder issued(Instant issued) {
+			property.issued = issued;
+			return this;
+		}
+
+		public Builder modified(Instant modified) {
+			property.modified = modified;
+			return this;
+		}
+
+		@JsonProperty("version")
+		public Builder version(Long version) {
+			property.version = version;
+			return this;
+		}
+
+		public ApplicationProperty build() {
+			if(property.key == null || property.key.isBlank()) {
```

Comment:
Key is defined as identifier, is this check redundant?
If it will create applicationProperty object and insert it into mongo with key=null or blank, maybe we can add @NotNull
You can also cover it with junit test, to see will it work as expected